README
art.gnome.org Version 3

This project's svn repository can be found in http://svn.gnome.org/viewcvs/art-web/branches/art-web-3/

The project derives from CodeIgniter PHP Framework (http://www.codeigniter.com).

The following directories and files contain arg.gnome.org Version 3 code:
	* /system/application
	* /assets
	* /db
	* /install.php
	


CodeIgniter code is under the /system directory except /system/application directory.

ChangeLog
2007-08-04  Benjamin Berg  <benjamin@sipsolutions.net>

	* includes/common.inc.php:
	Use theme:relation="$category" instead of "download".

2007-07-28  Benjamin Berg  <benjamin@sipsolutions.net>

	* includes/art_listings.inc.php:
	Add a where clause to top_ratings listing to show only active items.

2007-03-14  Thomas Wood  <thos@gnome.org>

	* includes/config.inc.php: Add GNOME 2.18 category to screenshots

2007-01-21  Thomas Wood  <thos@gnome.org>

	* users:
	* admin/edit_user.php

	Add a delete user function

2006-12-25 Lucas Lommer <llommer@cvs.gnome.org>

	* submit_screenshot.php:
	- new submission SQL query
	- missing screenshot date in db fixed
	- name and description quotes fix (no strip_string())

	* includes/art_headers.inc.php:
	- feed links removed from HTML head (no feeds for screenshots)

	* includes/art_page.inc.php:
	- screenshot release date output fixed (different column name in db)
	- do not create variations_list for screenshots, no parent column

2006-12-22 Lucas Lommer <llommer@cvs.gnome.org>

	* admin/show_submitted_themes.php:
	- theme name and description editing feature

	* admin/show_recent_comments.php:
	- use html_parse_text() for comments output

2006-12-18 Lucas Lommer <llommer@cvs.gnome.org>

	* default.css:
	- little CSS fix

	* includes/comments.inc.php:
	* templates/en/comments/preview.php:
	- comment preview time fix (mixed with edit preview)
	

2006-12-17  Thomas Wood  <thos@gnome.org>

	* includes/art_listings.inc.php: Include original parent item in
	variation listing

2006-12-17 Lucas Lommer <llommer@cvs.gnome.org>

	* ChangeLog:
	- My last entry date corrected

	* default.css:
	- Added comment_preview CSS class for comments preview
	- Added phpBB_box CSS class, a phpBB preview box
	- Some other new CSS related to previews

	* edit_background.php:
	- Added background description preview support
	- Added LIMIT 1 for SQL SELECT/UPDATE
	- Using strip_string() on strings going to preview
	- Using escape_string() on strings going to db
	- Partially done by Thomas Wood <thos@gnome.org>

	* edit_comment.php:
	- Added comment preview support
	- Added LIMIT 1 for SQL SELECT/UPDATE, do no use "SELECT *"
	- Using strip_string() on strings going to preview
	- Using escape_string() on strings going to db

	* edit_theme.php:
	- Added theme description preview support
	- Added LIMIT 1 for SQL SELECT/UPDATE
	- Using strip_string() on strings going to preview
	- Using escape_string() on strings going to db

	* submit_screenshot.php:
	- Using escape_string() on strings going to db

	* users:
	- Changed subtopic
	- We want to apply html_parse_text() on users' latest comments

	* admin/add_faq.php:
	- Added new FAQ preview support
	- Some HTML code is out of PHP now (faster)
	- Form look changes (to be same as on other sites)
	- Using escape_string() on strings going to db

	* admin/delete_faq.php:
	- Added width: 95%; inline CSS for select box (was too wide by default)

	* admin/edit_faq.php:
	- Added FAQ preview support
	- Some HTML code is out of PHP now (faster)
	- Has been rewritten for a bit :)
	- More changes in branch later (needs to be done differently)

	* includes/art_page.inc.php:
	- Added preview support for new comments
	- Code for adding comment into db came from includes/comments.inc.php

	* includes/comments.inc.php:
	- Added new comments preview support
	- get_comment_form() has two new attributes:
		$preview (false by default) - if true, function shows comment preview
		$error_fallback (false by default) - indicates if comment saving
									  produced an error, for fallback
	- get_comment_form() is using strip_string() on strings going to preview
	- add_comment() is using escape_string() on strings going to db
	- report_comment() is newly validating $commentID
	- Code for adding comment into db moved to includes/art_page.inc.php
	- Error messages are using global CSS styles

	* includes/common.inc.php:
	- Removed an empty line (I was going to do something there or what)

	* templates/en/edit_background.html:
	* templates/en/edit_comment.html:
	* templates/en/edit_theme.html:
	* templates/en/comments/add.html:
	- Textareas have own CSS class now
	- No save button, just preview (you have to see preview before saving)
	- Tabindexes should be finally done I hope

	* templates/en/edit_background_preview.html:
	* templates/en/edit_comment_preview.html:
	* templates/en/edit_theme_preview.html:
	* templates/en/comments/preview.html:
	- Newly added templates
	- Needed for preview support

2006-12-08  Thomas Wood  <thos@gnome.org>

	* includes/art_page.inc.php:
	* includes/common.inc.php:
	* templates/en/backgrounds/detail.html:
	* templates/en/contests/detail.html:
	* templates/en/screenshots/detail.html:
	* templates/en/themes/detail.html:

	Add 'breadcrumb' feature

2006-12-05 Lucas Lommer <llommer@cvs.gnome.org>

	* default.css:
	- Added some CSS classes used in forms

	* edit_background.php:
	* edit_theme.php:
	- We can set selectbox tabindex as new attribute

	* delete_comment.php:
	- Newly added file
	- Allows users to delete their own comments
	- Based on code in edit_comment.php

	* submit_background.php:
	- Check if new background name already exists in db
	- If name already exists, add 'Another ' prefix to background name

	* submit_theme.php:
	- Check if new theme name already exists in db
	- If name already exists, add 'Another ' prefix to theme name

	* includes/comments.inc.php:
	- Added 'Delete' link next to 'Edit' under user's comments
	- Users can delete own comments now using this link

	*includes/common.inc.php:
	- Function create_select_box() has new attribute $tabindex
	- Function create_select_box() allows you to set selectbox tabindex now

	* templates/en/delete_comment.html:
	- Newly added file
	- Template for comments deleting page
	- Based on code in templates/en/edit_comment.html

	* templates/en/edit_background.html:
	* templates/en/edit_comment.html:
	* templates/en/edit_theme.html:
	- Added missing tabindexes

xml.php
<?php
require("mysql.inc.php");
require("common.inc.php");

header("Content-type: text/plain; charset=ISO-8859-1");


//The Art:
//10 = Backgrounds Gnome
//11 = Backgrounds Other
//12 = Backgrounds All
//13 = Backgrounds Nature
//14 = Backgrounds Abstract

//20 = Desktop Themes Application
//21 = Desktop Themes Window Border
//22 = Desktop Themes Icons

//30 = Other Themes Login Manager
//31 = Other Themes Splash Screens
//32 = Other Themes Gtk+ Engines

$art = $_GET['art'];


function list_backgrounds($parent, $category)
{
	
	if ($category == "all")
		$sql = "SELECT * FROM background, user WHERE status = 'active' AND parent = $parent AND background.userID=user.userID";
	else
		$sql = "SELECT * FROM background, user WHERE status = 'active' AND parent = $parent AND category = '" . $category . "' AND background.userID=user.userID";


	$background_select_result = mysql_query($sql);

	while ($row = mysql_fetch_assoc($background_select_result))
	{
		$background_id = $row['backgroundID'];
		$background_name = $row['name'];
		$background_description = $row['background_description'];
		$background_category = $row['category'];
		$background_license = $row['license'];
		$background_vote_sum = $row['vote_sum'];
		$background_vote_count = $row['vote_count'];
		$background_release_date = $row['release_date'];
		$background_download_start_timestamp = $row['download_start_timestamp'];
		$background_download_count = $row['download_count'];
	
		$username = $row['username'];
		
		$thumbnail = $row['thumbnail_filename'];

		$archive = "";

		if ($background_id < 1000)
			$archive = "archive/";

		print("\t<background release_date=\"$background_release_date\" vote_sum=\"$background_vote_sum\" vote_count=\"$background_vote_count\" download_start_timestamp=\"$background_download_start_timestamp\" download_count=\"$background_download_count\">\n");
		print("\t\t<name>" . htmlspecialchars($background_name) . "</name>\n");
		print("\t\t<description>" . htmlspecialchars($background_description) . "</description>\n");
		print("\t\t<category>" . htmlspecialchars($background_category) . "</category>\n");
		print("\t\t<author>" . htmlspecialchars($username) . "</author>\n");
		print("\t\t<license>" . htmlspecialchars($background_license) . "</license>\n");

		print("\t\t<thumbnail>http://art.gnome.org/images/{$archive}thumbnails/backgrounds/$thumbnail</thumbnail>\n");

		foreach ($row as $key => $val)
		{
//			if ($key != "status" and $key != "parent")
//			print("\t\t<$key>$val</$key>\n");

		}

		$resolution_select_result = mysql_query("SELECT * FROM background_resolution WHERE backgroundID = '$background_id'");
		while ($res_row = mysql_fetch_assoc($resolution_select_result))
		{
			print("\t\t<background_resolution>\n");
			print("\t\t\t<type>{$res_row['type']}</type>\n");
			print("\t\t\t<resolution>{$res_row['resolution']}</resolution>\n");
			print("\t\t\t<url>http://art.gnome.org/download/backgrounds/{$category}/{$res_row['background_resolutionID']}/{$res_row['filename']}</url>\n");
			print("\t\t</background_resolution>\n");
		}

		list_backgrounds($background_id, $category);
		print("\t</background>\n");
	}
}




function list_themes($parent, $category)
{
	$sql = "SELECT * FROM theme, user WHERE status = 'active' AND parent = $parent AND category = '" . $category . "' AND theme.userID=user.userID";

	$theme_select_result = mysql_query($sql);

	while ($row = mysql_fetch_assoc($theme_select_result))
	{
		$theme_id = $row['themeID'];
		$theme_name = $row['name'];
		$theme_description = $row['description'];
		$theme_category = $row['category'];
		$username = $row['username'];
		$theme_license = $row['license'];
		$theme_vote_sum = $row['vote_sum'];
		$theme_vote_count = $row['vote_count'];
		$theme_download_start_timestamp = $row['download_start_timestamp'];
		$theme_download_count = $row['download_count'];
		$theme_release_date = $row['release_date'];
		
		$thumbnail = $row['preview_filename'];
		$small_thumbnail = $row['thumbnail_filename'];

		$archive = "";

		if ($theme_id < 1000)
			$archive = "archive/";

		print("\t<theme release_date=\"$theme_release_date\" vote_sum=\"$theme_vote_sum\" vote_count=\"$theme_vote_count\" download_start_timestamp=\"$theme_download_start_timestamp\" download_count=\"$theme_download_count\">\n");
		print("\t\t<name>" . htmlspecialchars($theme_name) . "</name>\n");
		print("\t\t<description>" . htmlspecialchars($theme_description) . "</description>\n");
		print("\t\t<category>" . htmlspecialchars($category) . "</category>\n");
		print("\t\t<author>" . htmlspecialchars($username) . "</author>\n");
		print("\t\t<license>" . htmlspecialchars($theme_license) . "</license>\n");
		print("\t\t<thumbnail>http://art.gnome.org/images/{$archive}thumbnails/{$theme_category}/{$thumbnail}</thumbnail>\n");
		print("\t\t<small_thumbnail>http://art.gnome.org/images/{$archive}thumbnails/{$theme_category}/{$small_thumbnail}</small_thumbnail>\n");
		print("\t\t<url>http://art.gnome.org/download/themes/{$theme_category}/{$theme_id}/{$row['download_filename']}</url>\n");
		foreach ($row as $key => $val)
		{
//			if ($key != "status" and $key != "parent")
//			print("\t\t<$key>$val</$key>\n");

		}


		list_themes($theme_id, $category);
		print("\t</theme>\n");
	}
}

print("<?xml version=\"1.0\" encoding=\"ISO-8859-1\" ?>\n");
print("<art>\n");


switch ($art) {
case 10:
	list_backgrounds(0, "gnome");
	break;
case 11: 
	list_backgrounds(0, "other");
	break;
case 12: 
	list_backgrounds(0, "all");
	break;
case 13:
	list_backgrounds(0, "nature");
	break;
case 14:
	list_backgrounds(0, "abstract");
	break;
case 20:
	list_themes(0, "gtk2");
	break;
case 21:
	list_themes(0, "metacity");
	break;
case 22:
	list_themes(0, "icon");
	break;
case 30:
	list_themes(0, "gdm_greeter");
	break;
case 31: 
	list_themes(0, "splash_screens");
	break;
case 32: 
	list_themes(0, "gtk_engines");
	break;
default:
	list_backgrounds(0, "gnome");
}


print("</art>\n");
?>
